File: insert-sort.js
/**
 * 遍历将无序区域的第一个插入到有序数组中（逐个向前交互到适合的位置）
 * 缩小无序数组的范围直到空集
 */
function sort(array, asc) {
    for (let i = 1; i < array.length; i++)
        for (let j = i; j > 0; j--) {
            if (array[j - 1] > array[j] && asc)
                [array[j], array[j - 1]] = [array[j - 1], array[j]];
            if (array[j - 1] < array[j] && !asc)
                [array[j], array[j - 1]] = [array[j - 1], array[j]];
        }
}
const array = [3, 4, 5, 2, 9];
sort(array, false);
console.log(array);

File: insert-sort.cpp
#include "iostream"
using namespace std;
/**
 * 将无序区域第一个逐个比较插入到前面有序区域的适合位置
 * 不断缩小遍历的无序数组，直到至空集
**/
void sort(int *array, int len, bool asc)
{
    for (int i = 1; i < len; i++)
        for (int j = i; j > 0; j--)
        {
            if (array[j - 1] > array[j] && asc)
                swap(array[j - 1], array[j]);
            if (array[j - 1] < array[j] && !asc)
                swap(array[j - 1], array[j]);
        }
}
void swap(int &a, int &b)
{
    int temp = a;
    a = b;
    b = temp;
}
int main()
{
    int len = 5;
    int array[5] = {3, 6, 9, 1, 4};
    sort(array, len, false);
    for (int i = 0; i < len; i++)
        cout << array[i] << endl;
}